Add Soulstone keyboard

Strech opened the pull request:
## Description

Add new keyboard `Soulstone`

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

None 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] ~My change requires a change to the documentation.~
- [x] ~I have updated the documentation accordingly.~
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] ~I have added tests to cover my changes.~
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

dunk2k commented on keyboards/strech/soulstone/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,81 @@
+        KC_LCTL,  KC_LALT,                      MO(2),    MO(1),            KC_SPC,   KC_RALT,                                KC_RGUI,  KC_RCTL
+    ),
+    [1] = LAYOUT(
+        KC_GRV,   KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,  KC_UP,    KC_TRNS,  KC_MUTE,  KC_VOLD,  KC_VOLU,
+        KC_CAPS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_LEFT,  KC_DOWN,  KC_RGHT,  KC_SCLN,  KC_BSLS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,                      KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,                                KC_TRNS,  KC_TRNS
+    ),
+    [2] = LAYOUT(
+        KC_TRNS,  KC_1,     KC_2,     KC_3,     KC_4,     KC_5,             KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_MINS,  KC_EQL,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_P4,    KC_P5,    KC_P6,    KC_LBRC,  KC_RBRC,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_P1,    KC_P2,    KC_P3,    KC_TRNS,  KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,                      KC_TRNS,  KC_TRNS,          KC_PENT,  KC_PDOT,                                KC_TRNS,  KC_TRNS
+    ),
+    [3] = LAYOUT(
+        KC_TRNS,  KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,            KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,                      KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,                                KC_TRNS,  KC_TRNS
```
```suggestion
KC_GRV,   _______,  _______,  _______,  _______,  _______,          _______,  _______,  KC_UP,    _______,  KC_MUTE,  KC_VOLD,  KC_VOLU,
        KC_CAPS,  _______,  _______,  _______,  _______,  _______,          _______,  KC_LEFT,  KC_DOWN,  KC_RGHT,  KC_SCLN,  KC_BSLS,
        _______,  _______,  _______,  _______,  _______,  _______,          _______,  _______,  _______,  _______,  _______,  _______,  _______,
        _______,  _______,                      _______,  _______,          _______,  _______,                                _______,  _______
    ),
    [2] = LAYOUT(
        _______,  KC_1,     KC_2,     KC_3,     KC_4,     KC_5,             KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_MINS,  KC_EQL,
        _______,  _______,  _______,  _______,  _______,  _______,          _______,  KC_P4,    KC_P5,    KC_P6,    KC_LBRC,  KC_RBRC,
        _______,  _______,  _______,  _______,  _______,  _______,          _______,  KC_P1,    KC_P2,    KC_P3,    _______,  _______,  _______,
        _______,  _______,                      _______,  _______,          KC_PENT,  KC_PDOT,                                _______,  _______
    ),
    [3] = LAYOUT(
        _______,  KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,            KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,
        _______,  _______,  _______,  _______,  _______,  _______,          _______,  _______,  _______,  _______,  _______,  _______,
        _______,  _______,  _______,  _______,  _______,  _______,          _______,  _______,  _______,  _______,  _______,  _______,  _______,
        _______,  _______,                      _______,  _______,          _______,  _______,                                _______,  _______
```

Strech replied:
Is `_______` preferable over `KC_TRNS`? Both are aliases for the same. Could you please elaborate on that?

dunk2k commented on keyboards/strech/soulstone/readme.md:
```diff
@@ -0,0 +1,28 @@
+# Soulstone
+
+![Soulstone](https://i.imgur.com/aQBsgZ9.jpg)
```
```suggestion
![Soulstone](https://i.imgur.com/aQBsgZ9h.jpg)
```

waffle87 commented on keyboards/strech/soulstone/info.json:
```diff
@@ -0,0 +1,85 @@
+                {"matrix": [0, 6], "x": 7, "y": 0, "w": 1, "h": 1, "label": "Y"},
+                {"matrix": [0, 7], "x": 8, "y": 0, "w": 1, "h": 1, "label": "U"},
+                {"matrix": [0, 8], "x": 9, "y": 0, "w": 1, "h": 1, "label": "I"},
+                {"matrix": [0, 9], "x": 10, "y": 0, "w": 1, "h": 1, "label": "O"},
+                {"matrix": [0, 10], "x": 11, "y": 0, "w": 1, "h": 1, "label": "P"},
+                {"matrix": [0, 11], "x": 12, "y": 0, "w": 1, "h": 1, "label": "Del"},
+                {"matrix": [0, 12], "x": 13, "y": 0, "w": 1, "h": 1, "label": "Bspc"},
+                {"matrix": [1, 0], "x": 0, "y": 1, "w": 1.25, "h": 1, "label": "Tab"},
+                {"matrix": [1, 1], "x": 1.25, "y": 1, "w": 1, "h": 1, "label": "A"},
+                {"matrix": [1, 2], "x": 2.25, "y": 1, "w": 1, "h": 1, "label": "S"},
+                {"matrix": [1, 3], "x": 3.25, "y": 1, "w": 1, "h": 1, "label": "D"},
+                {"matrix": [1, 4], "x": 4.25, "y": 1, "w": 1, "h": 1, "label": "F"},
+                {"matrix": [1, 5], "x": 5.25, "y": 1, "w": 1, "h": 1, "label": "G"},
+                {"matrix": [1, 6], "x": 7.25, "y": 1, "w": 1, "h": 1, "label": "H"},
+                {"matrix": [1, 7], "x": 8.25, "y": 1, "w": 1, "h": 1, "label": "J"},
+                {"matrix": [1, 8], "x": 9.25, "y": 1, "w": 1, "h": 1, "label": "K"},
+                {"matrix": [1, 9], "x": 10.25, "y": 1, "w": 1, "h": 1, "label": "L"},
+                {"matrix": [1, 10], "x": 11.25, "y": 1, "w": 1, "h": 1, "label": ";"},
+                {"matrix": [1, 12], "x": 12.25, "y": 1, "w": 1.5, "h": 1, "label": "Enter"},
+                {"matrix": [2, 0], "x": 0, "y": 2, "w": 1.75, "h": 1, "label": "L Shift"},
+                {"matrix": [2, 1], "x": 1.75, "y": 2, "w": 1, "h": 1, "label": "Z"},
+                {"matrix": [2, 2], "x": 2.75, "y": 2, "w": 1, "h": 1, "label": "X"},
+                {"matrix": [2, 3], "x": 3.75, "y": 2, "w": 1, "h": 1, "label": "C"},
+                {"matrix": [2, 4], "x": 4.75, "y": 2, "w": 1, "h": 1, "label": "V"},
+                {"matrix": [2, 5], "x": 5.75, "y": 2, "w": 1, "h": 1, "label": "`"},
+                {"matrix": [2, 6], "x": 6.75, "y": 2, "w": 1, "h": 1, "label": "B"},
+                {"matrix": [2, 7], "x": 7.75, "y": 2, "w": 1, "h": 1, "label": "N"},
+                {"matrix": [2, 8], "x": 8.75, "y": 2, "w": 1, "h": 1, "label": "M"},
+                {"matrix": [2, 9], "x": 9.75, "y": 2, "w": 1, "h": 1, "label": ","},
+                {"matrix": [2, 10], "x": 10.75, "y": 2, "w": 1, "h": 1, "label": "."},
+                {"matrix": [2, 11], "x": 11.75, "y": 2, "w": 1, "h": 1, "label": "/"},
+                {"matrix": [2, 12], "x": 12.75, "y": 2, "w": 1.25, "h": 1, "label": "R Shift"},
+                {"matrix": [3, 0], "x": 0, "y": 3, "w": 1.25, "h": 1, "label": "L Ctrl"},
+                {"matrix": [3, 1], "x": 1.25, "y": 3, "w": 1.25, "h": 1, "label": "L Alt"},
+                {"matrix": [3, 3], "x": 3.50, "y": 3, "w": 1.25, "h": 1, "label": "L GUI"},
+                {"matrix": [3, 4], "x": 4.75, "y": 3, "w": 2, "h": 1, "label": "Space"},
+                {"matrix": [3, 6], "x": 6.75, "y": 3, "w": 2.25, "h": 1, "label": "Space"},
+                {"matrix": [3, 8], "x": 9, "y": 3, "w": 1.25, "h": 1, "label": "Fn"},
+                {"matrix": [3, 11], "x": 11.50, "y": 3, "w": 1.25, "h": 1, "label": "R Alt"},
+                {"matrix": [3, 12], "x": 12.75, "y": 3, "w": 1.25, "h": 1, "label": "R Ctrl"}
```
```suggestion
                {"matrix": [0, 0], "x": 0, "y": 0, "label": "Esc"},
                {"matrix": [0, 1], "x": 1, "y": 0, "label": "Q"},
                {"matrix": [0, 2], "x": 2, "y": 0, "label": "W"},
                {"matrix": [0, 3], "x": 3, "y": 0, "label": "E"},
                {"matrix": [0, 4], "x": 4, "y": 0, "label": "R"},
                {"matrix": [0, 5], "x": 5, "y": 0, "label": "T"},
                {"matrix": [0, 6], "x": 7, "y": 0, "label": "Y"},
                {"matrix": [0, 7], "x": 8, "y": 0, "label": "U"},
                {"matrix": [0, 8], "x": 9, "y": 0, "label": "I"},
                {"matrix": [0, 9], "x": 10, "y": 0, "label": "O"},
                {"matrix": [0, 10], "x": 11, "y": 0, "label": "P"},
                {"matrix": [0, 11], "x": 12, "y": 0, "label": "Del"},
                {"matrix": [0, 12], "x": 13, "y": 0, "label": "Bspc"},
                {"matrix": [1, 0], "x": 0, "y": 1, "w": 1.25, "label": "Tab"},
                {"matrix": [1, 1], "x": 1.25, "y": 1, "label": "A"},
                {"matrix": [1, 2], "x": 2.25, "y": 1, "label": "S"},
                {"matrix": [1, 3], "x": 3.25, "y": 1, "label": "D"},
                {"matrix": [1, 4], "x": 4.25, "y": 1, "label": "F"},
                {"matrix": [1, 5], "x": 5.25, "y": 1, "label": "G"},
                {"matrix": [1, 6], "x": 7.25, "y": 1, "label": "H"},
                {"matrix": [1, 7], "x": 8.25, "y": 1, "label": "J"},
                {"matrix": [1, 8], "x": 9.25, "y": 1, "label": "K"},
                {"matrix": [1, 9], "x": 10.25, "y": 1, "label": "L"},
                {"matrix": [1, 10], "x": 11.25, "y": 1, "label": ";"},
                {"matrix": [1, 12], "x": 12.25, "y": 1, "w": 1.5, "label": "Enter"},
                {"matrix": [2, 0], "x": 0, "y": 2, "w": 1.75, "label": "L Shift"},
                {"matrix": [2, 1], "x": 1.75, "y": 2, "label": "Z"},
                {"matrix": [2, 2], "x": 2.75, "y": 2, "label": "X"},
                {"matrix": [2, 3], "x": 3.75, "y": 2, "label": "C"},
                {"matrix": [2, 4], "x": 4.75, "y": 2, "label": "V"},
                {"matrix": [2, 5], "x": 5.75, "y": 2, "label": "`"},
                {"matrix": [2, 6], "x": 6.75, "y": 2, "label": "B"},
                {"matrix": [2, 7], "x": 7.75, "y": 2, "label": "N"},
                {"matrix": [2, 8], "x": 8.75, "y": 2, "label": "M"},
                {"matrix": [2, 9], "x": 9.75, "y": 2, "label": ","},
                {"matrix": [2, 10], "x": 10.75, "y": 2, "label": "."},
                {"matrix": [2, 11], "x": 11.75, "y": 2, "label": "/"},
                {"matrix": [2, 12], "x": 12.75, "y": 2, "w": 1.25, "label": "R Shift"},
                {"matrix": [3, 0], "x": 0, "y": 3, "w": 1.25, "label": "L Ctrl"},
                {"matrix": [3, 1], "x": 1.25, "y": 3, "w": 1.25, "label": "L Alt"},
                {"matrix": [3, 3], "x": 3.50, "y": 3, "w": 1.25, "label": "L GUI"},
                {"matrix": [3, 4], "x": 4.75, "y": 3, "w": 2, "label": "Space"},
                {"matrix": [3, 6], "x": 6.75, "y": 3, "w": 2.25, "label": "Space"},
                {"matrix": [3, 8], "x": 9, "y": 3, "w": 1.25, "label": "Fn"},
                {"matrix": [3, 11], "x": 11.50, "y": 3, "w": 1.25, "label": "R Alt"},
                {"matrix": [3, 12], "x": 12.75, "y": 3, "w": 1.25, "label": "R Ctrl"}
```

waffle87 commented on keyboards/strech/soulstone/info.json:
```diff
@@ -0,0 +1,85 @@
+    "bootloader": "atmel-dfu",
+    "features": {
+        "bootmagic": true,
+        "mousekey": false,
```
This is preferred to be enabled by default, if possible.
```suggestion
        "mousekey": true,
```

drashna replied:
If space is an issue, enabling lto is pprobably a good idea.

Strech replied:
I will check the complication, thanks!

waffle87 commented on keyboards/strech/soulstone/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,80 @@
+        KC_TRNS,  KC_TRNS,                      KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,                                KC_TRNS,  KC_TRNS
+    )
+};
+
+void matrix_init_user(void) {
+    // Set CapsLock LED to output and low
+    setPinOutput(B1);
+    writePinLow(B1);
+    // Set NumLock LED to output and low
+    setPinOutput(B2);
+    writePinLow(B2);
+    // Set ScrollLock LED to output and low
+    setPinOutput(B3);
+    writePinLow(B3);
+}
+
+bool led_update_user(led_t led_state) {
+    if (led_state.caps_lock) {
+        writePinHigh(B1);
+    } else {
+        writePinLow(B1);
+    }
+    if (led_state.num_lock) {
+        writePinHigh(B2);
+    } else {
+        writePinLow(B2);
+    }
+    return false;
+}
```
```suggestion
```

waffle87 commented on keyboards/strech/soulstone/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,81 @@
+    )
+};
+
+void matrix_init_user(void) {
+    // Set CapsLock LED to output and low
+    setPinOutput(B1);
+    writePinLow(B1);
+    // Set NumLock LED to output and low
+    setPinOutput(B2);
+    writePinLow(B2);
+    // Set ScrollLock LED to output and low
+    setPinOutput(B3);
+    writePinLow(B3);
+}
+
+bool led_update_user(led_t led_state) {
+    if (led_state.caps_lock) {
+        writePinHigh(B1);
+    } else {
+        writePinLow(B1);
+    }
+    if (led_state.num_lock) {
+        writePinHigh(B2);
+    } else {
+        writePinLow(B2);
+    }
+    return false;
+}
+
+
```
```suggestion
```

Strech commented:
@waffle87 Thanks for the suggestions and the feedback, I've rebased my PR and addressed most comments. I've checked the compilation with the mouse feature, and everything is fine!

drashna commented on keyboards/strech/soulstone/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,54 @@
+
+// Function for layer indicator LED
+layer_state_t layer_state_set_user(layer_state_t state) {
+    if (get_highest_layer(state) == 1) {
+        writePinHigh(B3);
+    } else {
+        writePinLow(B3);
+    }
```
This might be better, as it will enable the LED if layer 1 is enabled, no matter if it's the highest layer or not.  Which sounds like it may be closer to what you want? 
```suggestion
    writePin(B3, layer_state_cmp(state, 1));
```

Strech replied:
I've replied here https://github.com/qmk/qmk_firmware/pull/22093#discussion_r1334962631

Strech commented on keyboards/strech/soulstone/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,56 @@
+    )
+};
+
+bool led_update_user(led_t led_state) {
+    writePin(B1, led_state.caps_lock);
+    writePin(B2, led_state.num_lock);
+    return false;
+}
```
@waffle87 Unfortunately, I have to return it in order to make B3 stay high while the non-default layer is active, otherwise it will briefly blink and turn off.

Strech commented on keyboards/strech/soulstone/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,56 @@
+}
+
+// Function for layer indicator LED
+layer_state_t layer_state_set_user(layer_state_t state) {
+    writePin(B3, !layer_state_cmp(state, 0));
+    return state;
+}
```
@drashna Sorry for the confusion, I was about to turn on B3 every time the non-default layer is active. I've used your approach to rewrite the code (also I've changed lines 46-50 to make it stay on, while layer is active)

Strech commented:
@waffle87 @drashna Could you please take another look? I've issues with LED on B3 to stay on with all suggestions, but I've managed to shrink the initial code with a better code while keeping it tiny. Thanks!

Strech commented:
Hey folks @waffle87 @drashna, is there anything I should do to make this PR merged?

fauxpark commented on keyboards/strech/soulstone/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,56 @@
+    )
+};
+
+bool led_update_user(led_t led_state) {
+    writePin(B1, led_state.caps_lock);
+    writePin(B2, led_state.num_lock);
+    return false;
+}
+
```
```suggestion
```

Strech replied:
If I do that, led on B3 will not stay on high when I press MO keys. I've tried it and tested it. That's the reason I've returned it.

It's the same code as here https://github.com/qmk/qmk_firmware/blob/master/keyboards/primekb/prime_e/keymaps/default/keymap.c#L60-L72

Strech replied:
I've made a short video on how it's manifesting. At the beginning - I'm using current PR firmware, and then flash it with suggested changes.

As you can see - instead of lightning it just briefly goes on and off.

https://github.com/qmk/qmk_firmware/assets/920053/be023035-6e0d-49d0-bcd7-cd41916e2f5b

Strech replied:
I hope that helps @fauxpark

fauxpark replied:
If this LED is not meant as a scroll lock indicator, then you should not set it as such in the config.h. That's why you're getting this behaviour.

Strech replied:
Thanks @fauxpark I have the same understanding, but when I remove the code as you suggested and also remove B3 as scroll lock LED, for some reason, this LED gets less power, here is a video

```diff
diff --git a/keyboards/strech/soulstone/info.json b/keyboards/strech/soulstone/info.json
index 0710dff957..53037d1460 100644
--- a/keyboards/strech/soulstone/info.json
+++ b/keyboards/strech/soulstone/info.json
@@ -23,9 +23,8 @@
         "rows": ["D0", "D4", "B5", "F5"]
     },
     "indicators": {
-        "num_lock": "B1",
-        "caps_lock": "B2",
-        "scroll_lock": "B3"
+        "caps_lock": "B1",
+        "num_lock": "B2"
     },
     "dynamic_keymap": {
         "layer_count": 6
```

and

```diff
diff --git a/keyboards/strech/soulstone/keymaps/default/keymap.c b/keyboards/strech/soulstone/keymaps/default/keymap.c
index 7e3ea67096..3f9691b7af 100644
--- a/keyboards/strech/soulstone/keymaps/default/keymap.c
+++ b/keyboards/strech/soulstone/keymaps/default/keymap.c
@@ -43,11 +43,11 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
     )
 };
 
-bool led_update_user(led_t led_state) {
-    writePin(B1, led_state.caps_lock);
-    writePin(B2, led_state.num_lock);
-    return false;
-}
+/* bool led_update_user(led_t led_state) { */
+/*     writePin(B1, led_state.caps_lock); */
+/*     writePin(B2, led_state.num_lock); */
+/*     return false; */
+/* } */
 
 // Function for layer indicator LED
 layer_state_t layer_state_set_user(layer_state_t state) {
```

https://github.com/qmk/qmk_firmware/assets/920053/ffd61141-f166-41d6-998d-1db3603b03fb

Strech replied:
I find that if I add `setPinOutput` to that function it shines with the same brightness

```diff
diff --git a/keyboards/strech/soulstone/keymaps/default/keymap.c b/keyboards/strech/soulstone/keymaps/default/keymap.c
index 7e3ea67096..4f2be37945 100644
--- a/keyboards/strech/soulstone/keymaps/default/keymap.c
+++ b/keyboards/strech/soulstone/keymaps/default/keymap.c
@@ -43,14 +43,15 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
     )
 };
 
-bool led_update_user(led_t led_state) {
-    writePin(B1, led_state.caps_lock);
-    writePin(B2, led_state.num_lock);
-    return false;
-}
- 
 // Function for layer indicator LED
 layer_state_t layer_state_set_user(layer_state_t state) {
+    setPinOutput(B3);
     writePin(B3, !layer_state_cmp(state, 0));
     return state;
 }
```

is it a correct way? @fauxpark

fauxpark commented on keyboards/strech/soulstone/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,56 @@
+    )
+};
+
+bool led_update_user(led_t led_state) {
+    writePin(B1, led_state.caps_lock);
+    writePin(B2, led_state.num_lock);
+    return false;
+}
+
```
```suggestion
```

Strech replied:
See commend above ⬆️

Strech commented:
@fauxpark Thanks a lot for the help, I've adjusted the code accordingly to your recommendation and added some required matrix init to have equal brightness for the layer LED.

drashna commented on keyboards/strech/soulstone/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,56 @@
+    )
+};
+
+void matrix_init_user(void) {
+    // Prepare layer indicator LED
+    setPinOutput(B3);
+    writePinLow(B3);
+}
+
+// Function for layer indicator LED
+layer_state_t layer_state_set_user(layer_state_t state) {
+    writePin(B3, !layer_state_cmp(state, 0));
+    return state;
+}
```
These should be moved to soulstone.c instead of here.  Same with the via keymap. 
```suggestion
```

The soulstone.c should look something like this: 

```c
// Copyright 2023 QMK
// SPDX-License-Identifier: GPL-2.0-or-later

#include "quantum.h"

void keyboard_post_init_kb(void) {
    // Prepare layer indicator LED
    setPinOutput(B3);
    writePinLow(B3);
    keyboard_post_init_user();
}

// Function for layer indicator LED
layer_state_t layer_state_set_kb(layer_state_t state) {
    writePin(B3, !layer_state_cmp(state, 0));
    return layer_state_set_user(state);
}
```

drashna commented on keyboards/strech/soulstone/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,56 @@
+        KC_TRNS,  KC_TRNS,                      KC_TRNS,  KC_TRNS,          KC_TRNS,  KC_TRNS,                                KC_TRNS,  KC_TRNS
+    )
+};
+
+void matrix_init_user(void) {
+    // Prepare layer indicator LED
+    setPinOutput(B3);
+    writePinLow(B3);
+}
+
+// Function for layer indicator LED
+layer_state_t layer_state_set_user(layer_state_t state) {
+    writePin(B3, !layer_state_cmp(state, 0));
+    return state;
+}
```
```suggestion
```

Strech commented:
@drashna thanks for the guidance, I've updated `keymap.c` and `soulstone.c`, tested it and it works as expected 👍🏼

fauxpark commented on keyboards/strech/soulstone/config.h:
```diff
@@ -0,0 +1,22 @@
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
```
```suggestion
#define LOCKING_RESYNC_ENABLE

#define LAYER_INDICATOR_LED_PIN B3
```

fauxpark commented on keyboards/strech/soulstone/soulstone.c:
```diff
@@ -0,0 +1,30 @@
+
+// Prepare layer indicator LED
+void keyboard_post_init_kb(void) {
+    setPinOutput(B3);
+    writePinLow(B3);
```
```suggestion
    setPinOutput(LAYER_INDICATOR_LED_PIN);
    writePinLow(LAYER_INDICATOR_LED_PIN);
```

fauxpark commented on keyboards/strech/soulstone/soulstone.c:
```diff
@@ -0,0 +1,30 @@
+
+// Function for layer indicator LED
+layer_state_t layer_state_set_kb(layer_state_t state) {
+    writePin(B3, !layer_state_cmp(state, 0));
```
```suggestion
    writePin(LAYER_INDICATOR_LED_PIN, !layer_state_cmp(state, 0));
```

Strech commented:
@fauxpark Thanks for the suggestions, I think now it looks more DRY 🧹

Strech commented:
@fauxpark 👋 Hey, is there anything else that holds it from being merged?
